sql/sql_table.cc

```diff
@@ -4080,15 +4082,19 @@ mysql_prepare_create_table(THD *thd, const char *error_schema_name,
                      MYF(0));
           DBUG_RETURN(TRUE);
         }
-        if ((create_info->db_type->db_type == DB_TYPE_TIANMU) &&
-            (!(thd->variables.sql_mode & MODE_NO_KEY_ERROR))) {
-          my_message(ER_TIANMU_NOT_SUPPORTED_FULLTEXT_INDEX,
-                     ER(ER_TIANMU_NOT_SUPPORTED_FULLTEXT_INDEX), MYF(0));
+        if (create_info->db_type->db_type == DB_TYPE_TIANMU){
+          sql_mode_t sql_mode = thd->variables.sql_mode;
```

Comment:
`sql_mode` assignment should be outside `if` condition.
Alse, rewrite code as follows:
```
sql_mode_t sql_mode = thd->slave_thread ? global_system_variables.sql_mode : thd->variables.sql_mode;
```

Reply:
What does it mean outside the if condition?